src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/caseupdate/services/CaseUpdater.java

```diff
@@ -27,18 +26,29 @@
 
     private static final Logger LOG = getLogger(CaseUpdater.class);
 
+    private final CaseUpdateRequestValidator caseUpdateRequestValidator;
+    private final UpdatedCaseValidator updatedCaseValidator;
     private final AddressExtractor addressExtractor;
 
-    public CaseUpdater(AddressExtractor addressExtractor) {
+    public CaseUpdater(
+        CaseUpdateRequestValidator caseUpdateRequestValidator,
+        UpdatedCaseValidator updatedCaseValidator,
+        AddressExtractor addressExtractor
+    ) {
+        this.caseUpdateRequestValidator = caseUpdateRequestValidator;
+        this.updatedCaseValidator = updatedCaseValidator;
         this.addressExtractor = addressExtractor;
     }
 
     public SuccessfulUpdateResponse update(CaseUpdateRequest caseUpdateRequest) {
+        // TODO remove this check after eliminating exception_record element from CaseUpdateRequest
         Assert.notEmpty(
             caseUpdateRequest.transformationInput.scannedDocuments,
             "Missing scanned documents in exception record"
         );
 
+        caseUpdateRequestValidator.assertIsValid(caseUpdateRequest);
+
```

Comment:
can we not just "trust"/"use" spring rest controller's built-in validation? we already build this request as follows:

```java
@Valid @RequestBody CaseUpdateRequest req
```

this will save all these unnecessary components/mocks in tests :(

Reply:
if it's not valid - it's not valid. immediate response without executing any of service functions